### PR TITLE
Update File based User Store properties

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Allow admin role for users defined in File-based User Store.
Fixes https://github.com/wso2/micro-integrator/issues/2356

## User stories

Set `user.is_admin` property to true to define the user as an admin user in the deployment.toml.

```toml
[[internal_apis.users]]
user.name = "admin"
user.password = "admin"
user.is_admin = true
```

### CODE REVIEW

File: components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java [hunk 1]
```diff
@@ -214,13 +215,19 @@ private static void populateUserStore(OMElement apiConfig) {
...
                     if (userNameElement != null && passwordElement != null) {
                         String userName = userNameElement.getText();
                         if (userMap.containsKey(userName)) {
                             handleException("Error parsing the file based user store. User: " + userName + " defined "
                                             + "more than once. ");
                         }
-                        userMap.put(userName, resolveSecret(passwordElement.getText()).toCharArray());
+                        boolean isAdmin = false;
+                        if (isAdminElement != null) {
+                            isAdmin = Boolean.parseBoolean(isAdminElement.getText());
```
Comment (Nirothipan) [inline components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java:227]: let's trim this and parse to boolean?

File: components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/UserInfo.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * This class is the DTO for User configs in internal-apis.xml
+ */
+public class UserInfo {
+
+    private char[] password;
+    private boolean admin;
+
+    public UserInfo(char[] password, boolean admin) {
```
Comment (Nirothipan) [inline components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/UserInfo.java:29]: Shall we include user name also in the DTO.

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/AuthorizationHandler.java [hunk 3]
```diff
@@ -82,12 +82,24 @@ protected Boolean authorize(String userName) {
...
     }
 
+    /**
+     * Processes the authorization request using file based user store.
+     *
+     * @return true if successfully authorized
+     */
+    private boolean processAuthorizationWithFileBasedUserStore(String userName) {
+
+        boolean isAuthorized = FileBasedUserStoreManager.getUserStoreManager().isAdmin(userName);
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/AuthorizationHandler.java:96]: here, the logic is we are authorizing if that is an admin, right? In that case, shall we rename isAuthorized ti isAdmin, else code is not readable.

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/SecurityUtils.java [hunk 4]
```diff
@@ -102,4 +104,20 @@ public static Boolean isFileBasedUserStoreEnabled() {
         }
         return false;
     }
+
+    /**
+     * Method to assert if a user is an admin
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/SecurityUtils.java:109]:
```suggestion
     * Method to assert if a user is an admin.
```

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/FileBasedUserStoreManager.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    private static void initializeUserStore() {
+
+        ManagementApiParser mgtParser = new ManagementApiParser();
+        try {
+            usersList = mgtParser.getUserMap();
+            isInitialized = true;
+        } catch (UserStoreUndefinedException e) {
+            LOG.error("User store config has not been defined in file "
+                    + ManagementApiParser.getConfigurationFilePath(), e);
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/FileBasedUserStoreManager.java:103]: why do we have a line break here, does this exceed 120 ?

File: components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java [hunk 6]
```diff
@@ -204,8 +205,8 @@ private static void populateUserStore(OMElement apiConfig) {
      * @param users the parent element of users
      * @return map of users against credentials
```
Comment (Nirothipan) [inline components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java:206]: shall we udpate this.

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/FileBasedUserStoreManager.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.inbound.endpoint.internal.http.api.UserInfo;
+import org.wso2.micro.integrator.management.apis.ManagementApiParser;
+import org.wso2.micro.integrator.management.apis.UserStoreUndefinedException;
+
+import java.util.Map;
+
+public class FileBasedUserStoreManager {
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/FileBasedUserStoreManager.java:30]: class level comments ?

Comment (SanojPunchihewa) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/FileBasedUserStoreManager.java:33]: Fixed with df6ab2b

File: components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java (hunk 6 above)
Comment (SanojPunchihewa) [reply components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java]: Fixed with df6ab2b

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/FileBasedUserStoreManager.java (hunk 5 above)
Comment (SanojPunchihewa) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/FileBasedUserStoreManager.java:106]: Yes

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/SecurityUtils.java (hunk 4 above)
Comment (SanojPunchihewa) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/SecurityUtils.java]: Fixed with df6ab2b

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/AuthorizationHandler.java (hunk 3 above)
Comment (SanojPunchihewa) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/AuthorizationHandler.java]: Fixed with df6ab2b

File: components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/UserInfo.java (hunk 2 above)
Comment (SanojPunchihewa) [reply components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/UserInfo.java]: Fixed with df6ab2b

File: components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java (hunk 1 above)
Comment (SanojPunchihewa) [reply components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java]: Fixed with df6ab2b

**Review by Nirothipan — APPROVED** [review]

**Review by Nirothipan — APPROVED** [review]

**Pull request merged**